raycicmd/converter.go

```diff
@@ -102,10 +101,81 @@ func (c *converter) convertGroup(g *pipelineGroup, filter *tagFilter) (
 func (c *converter) convertGroups(gs []*pipelineGroup, filter *tagFilter) (
 	[]*bkPipelineGroup, error,
 ) {
+
+	var groupNodes []*jobNode
+
+	for i, g := range gs {
+		node := &jobNode{
+			id:       fmt.Sprintf("g%d", i),
+			userKey:  g.Key,
+			srcGroup: g,
+			tags:     g.Tags,
+		}
+
+		for j, step := range g.Steps {
+			k, ok := stringInMap(step, "name")
+			if !ok {
+				k, _ = stringInMap(step, "key")
+			}
+
+			var tags []string
+			if v, ok := step["tags"]; ok {
+				tags = toStringList(v)
+			}
+
+			node.steps = append(node.steps, &jobNode{
+				id:      fmt.Sprintf("g%d_s%d", i, j),
+				userKey: k,
+				srcStep: step,
+				tags:    tags,
+			})
+		}
+
+		groupNodes = append(groupNodes, node)
+	}
+
+	// Build namedNodes, and check if we have duplicated user keys.
+	nameNodes := make(map[string]*jobNode)
```

Comment:
this map is used only to check duplicated keys and not used anywhere else? if so just need a set?

Reply:
done.

I will need something to query user defined names down the road.